tools/accuracy_checker/accuracy_checker/metrics/attribute_classification.py

```diff
@@ -0,0 +1,154 @@
+"""
+Copyright (c) 2019 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import numpy as np
+from sklearn.metrics import accuracy_score, precision_score, recall_score
+
+from ..config import BoolField, ListField
+from .metric import FullDatasetEvaluationMetric
+from ..representation import ContainerAnnotation, ContainerPrediction
+
+
+class AttributeClassificationMetric(FullDatasetEvaluationMetric):
+    """
+    Base metric class for evaluating metrics of attribute classification models.
+    """
+
+    annotation_types = (ContainerAnnotation, )
+    prediction_types = (ContainerPrediction, )
+
+    is_annotation_prediction_dict_computed = False
+    annotation_prediction_dict = {}
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'attributes': ListField(
+                optional=False,
+                description="List of attribute names."
+            ),
+            'calculate_average': BoolField(
+                optional=True, default=True, description="Allows calculation of average metric."
+            )
+        })
+        return parameters
+
+    def configure(self):
+        self.attributes = self.get_value_from_config('attributes')
+        self.calculate_average = self.get_value_from_config('calculate_average')
+        self.meta = self.__create_meta()
+        super().configure()
+
+    def evaluate(self, annotations, predictions):
+        pass
+
+    def submit_all(self, annotations, predictions):
+        return self.evaluate(annotations, predictions)
+
+    def compute_annotation_prediction_dict(self, annotations, predictions):
+        if self.is_annotation_prediction_dict_computed:
+            return
+        self.is_annotation_prediction_dict_computed = True
```

Comment:
please add reset method for reset annotation_prediction_dict, it is need for metric reusage in calibration process